### PR TITLE
[Improve] Flink cluster status monitoring improvement

### PR DESCRIPTION
**Pull request by wolfboys**
[Improve] Flink cluster status monitoring improvement

### CODE REVIEW

**Comment by wolfboys** [comment]
cc @xujiangfeng001 @RocMarshal PTAL, thx

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java [hunk 1]
```diff
@@ -321,11 +251,9 @@ private FinalApplicationStatus stringConvertFinalApplicationStatus(String value)
    * @return
    */
   private ClusterState finalApplicationStatusConvertClusterState(FinalApplicationStatus status) {
-    switch (status) {
-      case UNDEFINED:
-        return ClusterState.RUNNING;
-      default:
-        return ClusterState.STOPPED;
+    if (status == FinalApplicationStatus.UNDEFINED) {
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java:262]: When the yarn application was accepted by yarn resourcemanager, the FinalApplicationStatus is `UNDEFINED` too, but the flink cluster isn't `RUNNING`

This case was mentioned in https://github.com/apache/incubator-streampark/pull/2541#discussion_r1162899796

File: streampark-console/streampark-console-service/src/main/resources/mapper/core/ApplicationMapper.xml [hunk 2]
```diff
@@ -133,7 +133,7 @@
              limit 1
     </select>
 
-    <select id="getAffectedJobsByClusterId" resultType="java.lang.Integer" parameterType="java.lang.Long">
+    <select id="countJobsByClusterId" resultType="java.lang.Integer" parameterType="java.lang.Long">
         select
             count(1)
         from t_flink_app
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/resources/mapper/core/ApplicationMapper.xml:139]: Do we need to filter the status of the application here? https://github.com/apache/incubator-streampark/pull/2809#discussion_r1242367682

**Review by RocMarshal — COMMENTED** [review]
Thanks for @wolfboys @xujiangfeng001 driving this pr.
I left a few of comments, PTAL in your free time.
Thank you~

File: streampark-console/streampark-console-service/src/main/resources/mapper/core/ApplicationMapper.xml (hunk 2 above)
Comment (wolfboys) [reply streampark-console/streampark-console-service/src/main/resources/mapper/core/ApplicationMapper.xml:139]: > Do we need to filter the status of the application here? [#2809 (comment)](https://github.com/apache/incubator-streampark/pull/2809#discussion_r1242367682)

Thanks for your feedback. I agree with you. I think We should filter the status of the application

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java (hunk 1 above)
Comment (wolfboys) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java:262]: > When the yarn application was accepted by yarn resourcemanager, the FinalApplicationStatus is `UNDEFINED` too, but the flink cluster isn't `RUNNING`
> 
> This case was mentioned in [#2541 (comment)](https://github.com/apache/incubator-streampark/pull/2541#discussion_r1162899796)

I also have questions about this logic and further discussion is needed. @xujiangfeng001

File: streampark-console/streampark-console-service/src/main/assembly/script/schema/mysql-schema.sql [hunk 3]
```diff
@@ -438,7 +438,6 @@ drop table if exists `t_flink_cluster`;
 create table `t_flink_cluster` (
   `id` bigint not null auto_increment,
   `address` varchar(150) default null comment 'url address of cluster',
-  `job_manager_url` varchar(150) default null comment 'url address of jobmanager',
```
Comment (xujiangfeng001) [inline streampark-console/streampark-console-service/src/main/assembly/script/schema/mysql-schema.sql]: We need to confirm if we still need a ` job_ manager_ url`  to alleviate the request pressure of `resourceManager`.

File: streampark-console/streampark-console-service/src/main/resources/mapper/core/ApplicationMapper.xml (hunk 2 above)
Comment (xujiangfeng001) [reply streampark-console/streampark-console-service/src/main/resources/mapper/core/ApplicationMapper.xml:139]: > > Do we need to filter the status of the application here? [#2809 (comment)](https://github.com/apache/incubator-streampark/pull/2809#discussion_r1242367682)
> 
> Thanks for your feedback. I agree with you. I think We should filter the status of the application

It is difficult to directly query the data table for this issue. I think we need a cache to record the `running jobs` of each `flink cluster`.

**Comment by xujiangfeng001** [comment]
Hi @wolfboys @RocMarshal  ,I left a few of comments. Please take a look at it when you have time.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java (hunk 1 above)
Comment (xujiangfeng001) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java:262]: This is my negligence. The corresponding logic was not modified during the implementation of this piece of content. The logic here does not need to determine the `finalStatus `of the application, only the `application state` needs to be determined as `running` to consider the `flink cluster` as `running`.

```
public enum YarnApplicationState {
  /** Application which was just created. */
  NEW,

  /** Application which is being saved. */
  NEW_SAVING,

  /** Application which has been submitted. */
  SUBMITTED,

  /** Application has been accepted by the scheduler */
  ACCEPTED,

  /** Application which is currently running. */
  RUNNING,

  /** Application which finished successfully. */
  FINISHED,

  /** Application which failed. */
  FAILED,

  /** Application which was terminated by a user or admin. */
  KILLED
}
```

File: streampark-console/streampark-console-service/src/main/assembly/script/schema/mysql-schema.sql (hunk 3 above)
Comment (wolfboys) [reply streampark-console/streampark-console-service/src/main/assembly/script/schema/mysql-schema.sql]: > We need to confirm if we still need a ` job_ manager_ url` to alleviate the request pressure of `resourceManager`.

done

**Comment by wolfboys** [comment]
cc @RocMarshal @xujiangfeng001  PTAL

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java [hunk 4]
```diff
@@ -168,13 +135,29 @@ private void alert(FlinkCluster cluster, ClusterState state) {
    * @param flinkCluster
    * @return
    */
-  private ClusterState getClusterState(FlinkCluster flinkCluster) {
-    ClusterState state = getClusterStateFromFlinkAPI(flinkCluster);
-    if (ClusterState.isRunningState(state)) {
+  public ClusterState getClusterState(FlinkCluster flinkCluster) {
+    ClusterState state = FAILED_STATES.getIfPresent(flinkCluster.getId());
```
Comment (xujiangfeng001) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java:139]: I think there is a problem with the logic in this method: here, in Yarn Session mode, it won't go `jobmanagerUrl `request status, only `yarn restful api` requests will be processed. This is not in line with our original intention of designing the `jobmanagerUrl` field.

**Comment by xujiangfeng001** [comment]
Hi @wolfboys ，this generally looks good, I left a comment. If I make a mistake, please correct me.

**Review by RocMarshal — APPROVED** [review]
Thanks @wolfboys & @xujiangfeng001 .
LGTM +1.

**Pull request merged**